core/src/types/writer.rs

```diff
@@ -90,14 +91,22 @@ impl Writer {
         Ok(Writer { inner: w })
     }
 
-    /// Write into inner writer.
-    pub async fn write(&mut self, bs: impl Into<Bytes>) -> Result<()> {
-        let mut bs = bs.into();
+    pub async fn write(&mut self, bs: Buffer) -> Result<()> {
```

Comment:
Does this API meet expectations?

Reply:
I prefer to have `impl Into<Buffer>` for easier usage.